crates/polars-expr/src/reduce/bitwise.rs

```diff
@@ -0,0 +1,254 @@
+use std::ops::{BitAnd, BitOr, BitXor, Not};
+
+use arrow::array::BooleanArray;
+use arrow::types::NativeType;
+use num_traits::Zero;
+use polars_compute::bitwise::BitwiseKernel;
+use polars_core::with_match_physical_integer_polars_type;
+
+use super::*;
+use crate::reduce::min_max::{BoolMaxGroupedReduction, BoolMinGroupedReduction};
+
+pub fn new_bitwise_and_reduction(dtype: DataType) -> Box<dyn GroupedReduction> {
+    use DataType::*;
+    use VecMaskGroupedReduction as VMGR;
+    match dtype {
+        Boolean => Box::new(BoolMinGroupedReduction::default()),
+        _ if dtype.is_integer() => {
+            with_match_physical_integer_polars_type!(dtype.to_physical(), |$T| {
+                    Box::new(VMGR::new(dtype, NumReducer::<BitwiseAnd<$T>>::new()))
```

Comment:
I believe the formatter doesn't run in these macros; there is too much indentation.